wasi_snapshot_preview1/wasi_test.go

```diff
@@ -1909,93 +32,46 @@ func maskMemory(t *testing.T, ctx context.Context, mod api.Module, size int) {
 	}
 }
 
-func instantiateModule(ctx context.Context, t *testing.T, wasiFunction, wasiImport string, sysCtx *internalsys.Context) (api.Module, api.Function) {
-	r := wazero.NewRuntimeWithConfig(wazero.NewRuntimeConfigInterpreter())
+func requireModule(t *testing.T, config wazero.ModuleConfig) (api.Module, api.Closer, *bytes.Buffer) {
+	var log bytes.Buffer
 
-	_, err := Instantiate(testCtx, r)
-	require.NoError(t, err)
+	// Set context to one that has an experimental listener
+	ctx := context.WithValue(testCtx, FunctionListenerFactoryKey{}, NewLoggingListenerFactory(&log))
 
-	binary, err := watzero.Wat2Wasm(fmt.Sprintf(`(module
-  %[2]s
-  (memory 1 1)  ;; just an arbitrary size big enough for tests
-  (export "memory" (memory 0))
-  (export "%[1]s" (func $wasi.%[1]s))
-)`, wasiFunction, wasiImport))
-	require.NoError(t, err)
+	r := wazero.NewRuntimeWithConfig(wazero.NewRuntimeConfigInterpreter())
 
-	compiled, err := r.CompileModule(ctx, binary, wazero.NewCompileConfig())
+	compiled, err := (&builder{r}).moduleBuilder().
+		ExportMemoryWithMax("memory", 1, 1).
```

Comment:
one interesting thing is that I'm exporting memory in tests that use the wasi module. This is much simpler than the import/export dance.